src/include/storage/arrow_block_metadata.h

```diff
@@ -52,6 +94,12 @@ class ArrowBlockMetadata {
     return reinterpret_cast<ArrowVarlenColumn *>(null_count_end)[!col_id];
   }
 
+  ArrowDictColumn &GetDictColumn(const BlockLayout &layout, col_id_t col_id) {
```

Comment:
Realistically, I think we will need to merge the dict column with the varlen column class.